superset/db_engine_specs/presto.py

```diff
@@ -1385,10 +1390,53 @@ def get_extra_params(database: Database) -> Dict[str, Any]:
         })
 
         auth_header = r.headers.get("Authorization")
-        if auth_header:           
+        if auth_header:
+            auth_header = PrestoEngineSpec.__update_auth_header(auth_header)
+            
             session.headers["Authorization"] = auth_header
             connect_args["requests_session"] = session
             
         engine_params["connect_args"] = connect_args
         extra["engine_params"] = engine_params            
         return extra
+    
+    @staticmethod
+    def __update_auth_header(auth_header):
+        refresh_timeout = int(os.getenv('REFRESH_TIMEOUT', 600))
+        
+        token = auth_header.split(None, 1)
+        decoded_token = jwt.decode(token[1], options={"verify_signature": False})
```

Comment:
it is worth moving this inside if

Reply:
DOne

Reply:
Cool, please close this conversation, because I don't have permission to do this (there is no button).